### PR TITLE
Remove MSBuild and Mono from release packages

### PR DESCRIPTION
**Pull request by JoeRobich**
Closes https://github.com/OmniSharp/omnisharp-roslyn/issues/2339

### CODE REVIEW

**Comment by JoeRobich** [comment]
Getting mixed signals from the Mono test failures.

```
        [info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
                Located 1 MSBuild instance(s)
                            1: Mono 16.10.1 - "/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin"
        [dbug]: OmniSharp.CompositionHostBuilder
                MSBuild instance Mono 16.10.1 scored at 2
        [info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
                Registered MSBuild instance: Mono 16.10.1 - "/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin"
        [info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
                Loaded Microsoft.Build from '/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/Microsoft.Build.dll'
        [info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
                Loaded Microsoft.Build.Framework from '/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/Microsoft.Build.Framework.dll'
        [info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
                Loaded Microsoft.Build.Tasks.Core from '/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/Microsoft.Build.Tasks.Core.dll'
        [info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
                Loaded Microsoft.Build.Utilities.Core from '/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/Microsoft.Build.Utilities.Core.dll'
```

but then the exception is:

```
        System.TypeLoadException: Could not load file or assembly 'Microsoft.Build.Utilities.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies.
          at OmniSharp.MSBuild.ProjectFile.ProjectFileInfo.Load (System.String filePath, OmniSharp.MSBuild.ProjectIdInfo projectIdInfo, OmniSharp.MSBuild.ProjectLoader loader, System.Guid sessionId, OmniSharp.Services.DotNetInfo dotNetInfo) [0x00023] in <9e48cc6cdb05429387c2464a8132425b>:0 
          at OmniSharp.MSBuild.ProjectManager+<>c__DisplayClass32_0.<LoadProject>b__0 () [0x00000] in <9e48cc6cdb05429387c2464a8132425b>:0 
          at (wrapper delegate-invoke) System.Func`1[System.ValueTuple`3[OmniSharp.MSBuild.ProjectFile.ProjectFileInfo,System.Collections.Immutable.ImmutableArray`1[OmniSharp.MSBuild.Logging.MSBuildDiagnostic],OmniSharp.MSBuild.Notification.ProjectLoadedEventArgs]].invoke_TResult()
          at OmniSharp.MSBuild.ProjectManager.LoadOrReloadProject (System.String projectFilePath, System.Func`1[TResult] loader) [0x0001b] in <9e48cc6cdb05429387c2464a8132425b>:0 
```

**Comment by JoeRobich** [comment]
I believe I will need to learn more about this comment https://github.com/OmniSharp/omnisharp-roslyn/pull/2360/files#diff-c6eace5a70f0c2d920dfa97e988d4cad3d0131dd2b029bb2646c96222e866cf8L710-L711

**Comment by filipw** [comment]
At the moment the "pending" checks are appearing here because they are pinned by name in the branch protection rules, and since the names now changed due to version bumps, they appear as not-satisfied.

I do not believe there is a way to selectively apply different rule to this PR than it is to the other ones, so what I will suggest is that once we are ready to merge this, we merge using admin permissions without requirements met, and then adjust the branch protection rules once it's merged, as then all subsequent PRs will be built according to new task names.

File: src/OmniSharp.Host/MSBuild/Discovery/MSBuildLocator.cs [hunk 1]
```diff
@@ -34,33 +34,15 @@ protected override void DisposeCore(bool disposing)
 
         public static MSBuildLocator CreateDefault(ILoggerFactory loggerFactory, IAssemblyLoader assemblyLoader, IConfiguration msbuildConfiguration)
         {
-            var useBundledOnly = msbuildConfiguration.GetValue<bool>("UseBundledOnly");
```
Comment (filipw) [inline src/OmniSharp.Host/MSBuild/Discovery/MSBuildLocator.cs:37]: we could still look at this value and if we detect it to be `true`, we print a warning into the log saying that this is not supported anymore?

the way the configuration works, it does not even need to be part of the typed model anymore for us to be able to extract it manually here (like it is already done)

**Comment by filipw** [comment]
I think this looks great and removes a ton of complexities. I would just wait for us to successfully resolve the Nuget key issue and publish 1.38.2 before this is merged

File: src/OmniSharp.Host/MSBuild/Discovery/MSBuildLocator.cs [hunk 2]
```diff
@@ -34,33 +34,15 @@ protected override void DisposeCore(bool disposing)
...
-            if (useBundledOnly)
-            {
-                var logger = loggerFactory.CreateLogger<MSBuildLocator>();
-                logger.LogInformation("Because 'UseBundledOnly' is enabled in the configuration, OmniSharp will only use the bundled MSBuild.");
-                return CreateStandAlone(loggerFactory, assemblyLoader);
-            }
-
             return new MSBuildLocator(loggerFactory, assemblyLoader,
```
Comment (JoeRobich) [inline src/OmniSharp.Host/MSBuild/Discovery/MSBuildLocator.cs:37]:
```suggestion
            var useBundledOnly = msbuildConfiguration.GetValue<bool>("UseBundledOnly");
            if (useBundledOnly)
            {
                logger.LogWarning("The MSBuild option 'UseBundledOnly' is no longer supported. Please update your OmniSharp configuration files.");
            }

            return new MSBuildLocator(loggerFactory, assemblyLoader,
```

File: src/OmniSharp.Host/MSBuild/Discovery/MSBuildLocator.cs (hunk 1 above)
Comment (JoeRobich) [reply src/OmniSharp.Host/MSBuild/Discovery/MSBuildLocator.cs:37]: Good idea. Updated with a warning.

**Review by filipw — APPROVED** [review]
looks great!

**Pull request merged**